### ISSUE TITLE
problem with npm completion

### ISSUE THREAD

**Issue by MunifTanjim** [Bash-it/bash-it#811]
### Description
The npm completion depends on the `npm` command itself. The `npm completion` command generates the script for completion which is sourced to enable the completion.

But because, the `*.plugin.bash` files loads after loading the `*.completion.bash` files, the following problem occurs:

The `npm.completion.bash` file loads, but since the `nvm.plugin.bash` file is not loaded yet, the `npm` command is not found, so npm completion is not enabled.

### Suggested Solution
Change the following piece of code from `bash_it.sh` to load the plugins first:

```bash
# Load enabled aliases, completion, plugins
for file_type in "aliases" "completion" "plugins"
do
  _load_bash_it_files $file_type
done
```

**Comment by edubxb** [comment]
PR are welcome :wink:

**Comment by nwinkler** [comment]
The question is whether this will have any side effects...

This addresses a more general problem as well: Since there are no dependencies between plugins, and the order is purely alphabetical, the `nvm` plugin is loaded after the `node` plugin, which relies on `npm` being present on the path.

**Comment by MunifTanjim** [comment]
I'm not 100% sure, but I don't think there will be any side effects.
Should I create the PR now or wait?

**Comment by MunifTanjim** [comment]
Just noticed, the same problem applies to nvm completion as well. Loading plugins before completions solves that too.

**Comment by edubxb** [comment]
Loading plugins before completion make sense for me, but as @nwinkler points out, the lack of dependencies between plugins is the real problem.

IMO, this change is welcome.

**Comment by nwinkler** [comment]
Fixed in #813 - let me know if this works now!
